lib/Dialect/AIEX/Transforms/AIEDmaToNpu.cpp

```diff
@@ -634,10 +675,10 @@ struct WriteBdToBlockWritePattern : OpConversionPattern<NpuWriteBdOp> {
       std::string name = "blockwrite_data_";
       rewriter.setInsertionPoint(
           op->getParentOfType<AIEX::RuntimeSequenceOp>());
-      int id = 0;
-      while (dev.lookupSymbol(name + std::to_string(id)))
-        id++;
-      name += std::to_string(id);
```

Comment:
Maybe the issue was that it starts at zero each time when it doesn't have to?  If `id` was a static member variable of `WriteBdToBlockWritePattern`, initialized once to zero, it would probably have the same effect as this PR.

Reply:
Good point. PR updated. Thanks for the suggestion, @fifield!